examples/utils.cpp

```diff
@@ -1,6 +1,15 @@
 #include "utils.hpp"
 
+#if !((defined(__cplusplus) && (__cplusplus >= 201703L)) || (defined(_MSVC_LANG) && (_MSVC_LANG >= 201703L)))
```

Comment:
How about introduce a file `include/hyperpose/filesystem` and put all the `#if ...` in side it.
So all other files in this project can just `#include <hyperpose/filesystem>`.

e.g. 

define a customized  `filesystem` header
https://github.com/stdml/stdtensor/blob/master/tests/include/ttl/filesystem

and use it in another file without the `#if...` macros.
https://github.com/stdml/stdtensor/blob/ef1a5249fa04880e20bcd18df69c2d73ede41db8/tests/test_loc.cpp#L3

Reply:
Could do but I wonder if any of this is really necessary, if all the required C++ compilers (and versions of them) for hyperpose has C++17 standard conformance and C++17 is the current standard may aswell just not use the std::experimental version of filesystem.